README.md

```diff
@@ -118,3 +101,171 @@ Remote debugging using :2000
+
+ * libusb
+ * libftdi
+ * evt
+ * libhidapi
+
+### Building
+
+The project is configured and built using the meson buildsystem, you will need to create a build directory and then configure the build depending on what you want.
+
+The project has a couple of build options you may configure, which affect the resulting binaries, this is how you will configure the firmware for you respective probe, or enable certain features.
+
+A non-exhaustive list of project options:
+ * probe: Hardware platform where the BMD firmware will run
+ * targets: Enabled debug targets
+ * debug_output: Enable debug output (for debugging the BMD stack, not debug targets)
+ * rtt_support: Enable RTT (Real Time Transfer) support
+
+You may see all available project options and valid values under  `Project options` in the output of the `meson configure` command.
+
+The first configuration commands are ran from the root of your clone of the blackmagic repository
+``` sh
+cd /path/to/repository/blackmagic
+```
+
+All following commands are ran from inside the build directory.
+
+#### Building Black Magic Debug Firmware
+
+To build the firmware you need to configure the probe hardware the firmware will run on, as well as a cross-compilation toolchain used.
+
+For convenience, a cross-file for each supported hardware probe is available which provides a sane default configuration for it.
+
+The build configuration command for the native probe may look like:
+``` sh
+meson setup build --cross-file cross-file/native.ini
+```
+
+Alternatively (for advanced users), if you wish to configure manually, for instance while writing support for a new probe, or a different toolchain, you can run:
+``` sh
+meson setup build --cross-file cross-file/arm-none-eabi.ini -Dprobe=native
+```
+
+You now should have a `build` directory from where you can build the firmware, this is also where your binary files will appear.
+
+The command `meson compile` will build the default targets:
+```sh
+cd build
+meson compile
+```
+
+Ypu should now see the resulting binaries in `build`, in this case:
+```
+blackmagic_native_firmware.bin
+blackmagic_native_firmware.elf
+```
+
+These are the binary files you will use to flash to your probe.
+
+##### region `rom' overflowed
+
+It may happen, while working with non default configurations or the project latest version from git, that the firmware does not fit in the available space for the configured probe, this could look something like:
+```
+arm-none-eabi/bin/ld: region `rom' overflowed by 4088 bytes
+Memory region         Used Size  Region Size  %age Used
+             rom:      135160 B       128 KB    103.12%
+             ram:        5708 B        20 KB     27.87%
+collect2: error: ld returned 1 exit status
+```
+
+This is not unexpected, as some hardware probe have limited space for firmware, you can get around it by disabling some features or target support:
+``` sh
+meson setup --reconfigure -Dtargets=cortexm,stm -Drtt_support=false
```

Comment:
`meson configure -Dtargets=[...]` is preferred  (by us) honestly - and iwrc this is internally rewritten to this form. Trying to describe to users why they're having to do this wordier way is going to be troublesome.

Reply:
That would be the more correct way to do this, and is my preference too; But while testing the instructions to make sure everything was correct, I noticed that for some reason it did not seem to apply the change properly, the summary would still display the old value...
I did not have time to dive into it, if it's a bug with meson in general, or with the summary feature (or a misshap on my side), but I opted for the command that does work instead, at least for now until we can verify the behavior (If you could check on your system it would be welcome).

Reply:
We have tested on our system and it works perfectly for us - though we are on Meson 1.0.1 right now, so it's possible there's a bug in a newer Meson. Think we shouldn't be trying to work around that though and should write things the correct way here and report a bug instead. Let's fix the tool in preference 🙂

Reply:
Ah, great, I was afraid it might be something that affected older versions too, I'm on v1.3.1, I'll investigate this further, but revert these to the more correct command in the meantime